conda_libmamba_solver/state.py

```diff
@@ -732,26 +770,35 @@ def _prepare_for_add(self, index: IndexHelper):
-            elif name in explicit_pool:
-                # TODO: This might be introducing additional specs into the list if the pin
-                # matches a dependency of a request, but that dependency only appears in _some_
-                # of the request variants. For example, package A=2 depends on B, but package
-                # A=3 no longer depends on B. B will be part of A's explicit pool because it
-                # "could" be a dependency. If B happens to be pinned but A=3 ends up being the
-                # one chosen by the solver, then B would be included in the solution when it
-                # shouldn't. It's a corner case but it can happen so we might need to further
-                # restrict the explicit_pool to see. The original logic in the classic solver
-                # checked: `if explicit_pool[s.name] & ssc.r._get_package_pool([s]).get(s.name,
-                # set()):`
+            # always assume the pin will be needed
+            # elif name in explicit_pool:
+            # THIS BLOCK HAS BEEN DEACTIVATED
+            # the explicit pool is potentially expensive and we are not using it.
+            # leaving this here for reference. It's supposed to check whether the pin
+            # was going to be part of the environment because it shows up in the dependency
+            # tree of the explicitly requested specs.
+            # ---
+            # TODO: This might be introducing additional specs into the list if the pin
+            # matches a dependency of a request, but that dependency only appears in _some_
+            # of the request variants. For example, package A=2 depends on B, but package
+            # A=3 no longer depends on B. B will be part of A's explicit pool because it
+            # "could" be a dependency. If B happens to be pinned but A=3 ends up being the
+            # one chosen by the solver, then B would be included in the solution when it
+            # shouldn't. It's a corner case but it can happen so we might need to further
+            # restrict the explicit_pool to see. The original logic in the classic solver
+            # checked: `if explicit_pool[s.name] & ssc.r._get_package_pool([s]).get(s.name,
+            # set()):`
+            else:  # always add the pin for libmamba to consider it
                 self.specs.set(
                     name,
                     pin,
                     reason="Pin matches one of the potential dependencies of user requests",
                 )
-            else:
-                log.warn(
-                    "pinned spec %s conflicts with explicit specs. Overriding pinned spec.", spec
-                )
+            # In classic, this would notify the pin was being overridden by a request
+            # else:
+            #     log.warn(
+            #         "pinned spec %s conflicts with explicit specs. Overriding pinned spec.", spec
+            #     )
```

Comment:
Do we need this commented out?

Reply:
Yep, it's been replaced by the `else` clause above. This part of the code is not really used and needs to be removed anyway. But we'll clean up in a new PR after this one. I want this around for a release or two in case we need to debug some stuff.